### PR TITLE
feat: add task budget schema and CRUD functions

### PR DESCRIPTION
**Pull request by chapel**
## Summary

- Added `TaskBudgetSchema` to `src/sessions/types.ts` with `max_per_cycle` (positive int) and `started_this_cycle` (nonneg int) fields
- Added budget CRUD functions to `src/sessions/store.ts`: `createBudget()`, `getBudget()`, `checkBudget()`, `incrementBudget()`, `resetBudget()`
- Budget state stored in local filesystem at `.kspec/sessions/{id}/budget.json` (NOT shadow branch) to avoid contention between ralph and spawned agents
- Atomic writes via write-to-temp-then-rename pattern for crash safety
- 32 tests covering schema validation, all CRUD operations, enforcement logic, atomic write guarantees, and edge cases

## Test plan

- [x] Schema validates correctly (positive ints, rejects negatives/floats/missing fields)
- [x] Budget path resolves to correct local filesystem location
- [x] Create → read → increment → check → reset → check lifecycle works
- [x] Budget exhaustion blocks with clear message including counts (e.g., "1/1 tasks started")
- [x] No budget / no session → allowed (backward compatible, opt-in)
- [x] Corrupt/invalid budget.json treated as no budget (graceful degradation)
- [x] No .tmp files left after atomic writes
- [x] Full test suite passes (3321/3321, 1 pre-existing timeout in batch-exec-engine)

Task: @extend-session-schema-with-task-budget
Spec: @session-creation-and-env-injection

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/sessions/store.ts [hunk 1]
```diff
@@ -1524,3 +1538,185 @@ export async function searchSessionEvents(
...
+export async function getBudget(
+  specDir: string,
+  sessionId: string,
+): Promise<TaskBudget | null> {
+  const budgetPath = getSessionBudgetPath(specDir, sessionId);
+  try {
+    const content = await fsPromises.readFile(budgetPath, "utf-8");
+    const raw = JSON.parse(content);
+    return TaskBudgetSchema.parse(raw);
+  } catch {
```
Comment (chapel) [inline src/sessions/store.ts]: MUST-FIX: this broad `catch` turns every failure mode into `null` (missing file, malformed JSON, schema validation failure, permissions/I/O errors). Because `checkBudget()` treats `null` as "no budget", a corrupt/unreadable `budget.json` silently disables enforcement. `ac-no-budget` only specifies the missing-file case. Please narrow this to expected "not configured" errors (e.g. `ENOENT`) and surface malformed/unreadable budgets as errors so task-start fails closed instead of bypassing the limit.

File: tests/session-budget.test.ts [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+      expect(budget!.max_per_cycle).toBe(5);
+      expect(budget!.started_this_cycle).toBe(0);
+    });
+
+    it("should return null for nonexistent session", async () => {
+      const budget = await getBudget(testDir, "NONEXISTENT");
+      expect(budget).toBeNull();
+    });
+
+    it("should return null for corrupt budget.json (intentional: treated as no budget)", async () => {
```
Comment (chapel) [inline tests/session-budget.test.ts]: SHOULD-FIX (paired with the `getBudget()` change): this test locks in the fail-open behavior by treating corrupt `budget.json` as equivalent to no budget. That makes it easy for enforcement to be bypassed by file corruption. Once `getBudget()` distinguishes `ENOENT` from malformed/unreadable content, this should expect an error (or a dedicated invalid-budget result) instead of `null`.
